Zend/zend_language_scanner.l

```diff
@@ -1775,7 +1775,7 @@ NEWLINE ("\r"|"\n"|"\r\n")
 	char *end, *bin = yytext + 2;
 
 	/* Skip any leading 0s */
-	while (*bin == '0' || *bin == '_') {
+	while ((*bin == '0' || *bin == '_') && len > 0) {
```

Comment:
Please invert these conditions (i.e. check len first, then dereference). It doesn't matter here, but generally dereferencing things past the length is a bad idea.